packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/layoutanimation/AbstractLayoutAnimation.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.uimanager.layoutanimation
+
+import android.view.View
+import android.view.animation.AccelerateDecelerateInterpolator
+import android.view.animation.AccelerateInterpolator
+import android.view.animation.Animation
+import android.view.animation.BaseInterpolator
+import android.view.animation.DecelerateInterpolator
+import android.view.animation.Interpolator
+import android.view.animation.LinearInterpolator
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.common.annotations.VisibleForTesting
+import com.facebook.react.common.annotations.internal.LegacyArchitecture
+import com.facebook.react.common.annotations.internal.LegacyArchitectureLogLevel
+import com.facebook.react.common.annotations.internal.LegacyArchitectureLogger
+import com.facebook.react.uimanager.IllegalViewOperationException
+
+/**
+ * Class responsible for parsing and converting layout animation data into native [Animation]
+ * in order to animate layout when a valid configuration has been supplied by the application.
+ */
+@LegacyArchitecture
+internal abstract class AbstractLayoutAnimation {
+  @VisibleForTesting var interpolator: Interpolator? = null
+  @VisibleForTesting var delayMs: Int = 0
+  @VisibleForTesting var animatedProperty: AnimatedPropertyType? = null
+  @VisibleForTesting var durationMs: Int = 0
```

Comment:
I think the `@VisibleForTesting` is not needed here at all

Reply:
Removed in 79c69807b5da4913308cb62db97bf62c65eab84a

it's still a bit confusing to me when this should be used 😅